### PR TITLE
Add  com.github.tmewett.BrogueCE

### PR DESCRIPTION
**Pull request by adamj-codethink**
This adds a flatpak for https://github.com/tmewett/BrogueCE

I have opened an issue with the community asking for their approval and offered my help to update the app when a new release comes out.

https://github.com/tmewett/BrogueCE/issues/89

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.github.tmewett.BrogueCe

**Comment by LeandroStanger** [comment]
bot, build com.github.tmewett.BrogueCe

**Comment by LeandroStanger** [comment]
bot, build com.github.tmewett.BrogueCe

**Comment by barthalion** [comment]
The manifest file misses file extension.

File: com.github.tmewett.BrogueCE [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+    "app-id": "com.github.tmewett.BrogueCE",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "19.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "brogue",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
```
Comment (barthalion) [inline com.github.tmewett.BrogueCE:9]: Should be `fallback-x11`.

File: brogue.desktop [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Type=Application
+Name=Brogue
+GenericName=Roguelike
+Comment=Brave the Dungeons of Doom!
+Exec=brogue
+Icon=com.github.tmewett.BrogueCE
+Terminal=false
+Categories=Games;
+StartupNotify=true
+X-Flatpak=com.github.tmewett.BrogueCE
```
Comment (barthalion) [inline brogue.desktop:12]: I don't think this is needed?

File: com.github.tmewett.BrogueCE.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
```
Comment (barthalion) [inline com.github.tmewett.BrogueCE.appdata.xml:2]: The file misses OARS and developer name.

Comment (adamj-codethink) [reply com.github.tmewett.BrogueCE.appdata.xml:2]: Ah yes i will generate one of those also

**Comment by adamj-codethink** [comment]
@barthalion what do you mean, by misses extension sorry?

**Comment by bilelmoussaoui** [comment]
> @barthalion what do you mean, by misses extension sorry?

the flatpak manifest is named `com.github.tmewett.BrogueCE` instead of `com.github.tmewett.BrogueCE.json`

**Comment by adamj-codethink** [comment]
@bilelmoussaoui yes haha i just spotted that /o\ Fixed!

**Comment by bilelmoussaoui** [comment]
bot, build  com.github.tmewett.BrogueCE

**Comment by bilelmoussaoui** [comment]
bot,build com.github.tmewett.BrogueCE

**Comment by bilelmoussaoui** [comment]
bot, build com.github.tmewett.BrogueCE

**Comment by adamj-codethink** [comment]
@bilelmoussaoui any idea why it's trying to stat a gzip file in the builds?

File: com.github.tmewett.BrogueCE.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            "name": "brogue",
+            "buildsystem": "simple",
+            "build-commands": [
+                "export DATADIR=/app/lib/brogue/",
+                "make DATADIR=/app/lib/brogue/",
+                "install -Dm755 bin/brogue /app/lib/brogue/brogue",
+                "install -Dm755 brogue /app/bin/brogue",
+                "cp -r bin/assets /app/lib/brogue/assets",
+                "install -Dm644 bin/assets/icon.png /app/share/icons/hicolor/256x256/apps/com.github.tmewett.BrogueCE.png",
+                "install -Dm644 brogue.desktop /app/share/applications/com.github.tmewett.BrogueCE.desktop"
```
Comment (bilelmoussaoui) [inline com.github.tmewett.BrogueCE.json:25]: The metainfo file should be installed here

Comment (adamj-codethink) [reply com.github.tmewett.BrogueCE.json]: Ah yes, thanks i have fixed that also, fingers crossed now ha!

**Comment by LeandroStanger** [comment]
bot, build com.github.tmewett.BrogueCE

File: com.github.tmewett.BrogueCE.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            "buildsystem": "simple",
+            "build-commands": [
+                "export DATADIR=/app/lib/brogue/",
+                "make DATADIR=/app/lib/brogue/",
+                "install -Dm755 bin/brogue /app/lib/brogue/brogue",
+                "install -Dm755 brogue /app/bin/brogue",
+                "cp -r bin/assets /app/lib/brogue/assets",
+                "install -Dm644 bin/assets/icon.png /app/share/icons/hicolor/256x256/apps/com.github.tmewett.BrogueCE.png",
+                "install -Dm644 brogue.desktop /app/share/applications/com.github.tmewett.BrogueCE.desktop",
+                "install -Dm644 com.github.tmewett.BrogueCE.appdata.xml /app/share/data/com.github.tmewett.BrogueCE.appdata.xml"
```
Comment (bilelmoussaoui) [inline com.github.tmewett.BrogueCE.json:26]:
```suggestion
                "install -Dm644 com.github.tmewett.BrogueCE.metainfo.xml /app/share/metainfo/com.github.tmewett.BrogueCE.metainfoxml"
```
If you don't mind renaming the appdata file to use .metainfo.xml; it's the recommended directory to use per the specs. The appdata one is deprecated

Comment (adamj-codethink) [reply com.github.tmewett.BrogueCE.json:26]: Still  seeing the same error in the builds, could this be due to the names?

**Comment by bilelmoussaoui** [comment]
bot, build com.github.tmewett.BrogueCE

**Comment by bilelmoussaoui** [comment]
bot, build com.github.tmewett.BrogueCE

File: com.github.tmewett.BrogueCE.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>com.github.tmewett.BrogueCE</id>
+    <launchable type="desktop-id">com.github.tmewett.BrogueCE.desktop</launchable>
+    <name>BrogueCE</name>
+    <developer_name>BrogueCE</developer_name>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LGPL-3.0+</project_license>
```
Comment (tmewett) [inline com.github.tmewett.BrogueCE.metainfo.xml:8]: Brogue is AGPL 3.0

**Comment by tmewett** [comment]
Thanks for distributing this. How does Flathub work w.r.t responsibilities? Like am I ok to leave it to other people to maintain this distribution a la typical Linux packages?

File: com.github.tmewett.BrogueCE.metainfo.xml (hunk 6 above)
Comment (adamj-codethink) [reply com.github.tmewett.BrogueCE.metainfo.xml:8]: I will update this

**Comment by adamj-codethink** [comment]
@tmewett no problem, somebody is tagged as the owner of this repo and can update the app whenever it is needed, every 1/2 years we may also need to update to latest runtime, thats about it really

**Comment by barthalion** [comment]
bot, build com.github.tmewett.BrogueCE

**Comment by barthalion** [comment]
```
builddir/files/share/appdata/com.github.tmewett.BrogueCE.appdata.xml: FAILED:
• tag-missing           : <release> required
• tag-missing           : <description> required
```

I suppose it does not consider current `<description>` because it doesn't have any `<p>` tags. It's probably the best to generate skeleton with https://bilelmoussaoui.github.io/metainfo-generator/#/ or https://www.freedesktop.org/software/appstream/metainfocreator/#/ instead.

File: brogue.desktop [hunk 7]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=Brogue
+GenericName=Roguelike
+Comment=Brave the Dungeons of Doom!
+Exec=brogue
+Icon=com.github.tmewett.BrogueCE
+Terminal=false
+Categories=Games;
```
Comment (barthalion) [inline brogue.desktop:10]:
```
WARNING: Failed to validate desktop file /var/home/bpiotrowski/tmp/com.github.tmewett.BrogueCE/builddir/export/share/applications/com.github.tmewett.BrogueCE.desktop: /var/home/bpiotrowski/tmp/com.github.tmewett.BrogueCE/builddir/export/share/applications/com.github.tmewett.BrogueCE.desktop: error: value "Games;" for key "Categories" in group "Desktop Entry" contains an unregistered value "Games"; values extending the format should start with "X-"
/var/home/bpiotrowski/tmp/com.github.tmewett.BrogueCE/builddir/export/share/applications/com.github.tmewett.BrogueCE.desktop: hint: value "Games;" for key "Categories" in group "Desktop Entry" does not contain a registered main category; application might only show up in a "catch-all" section of the application menu
```

It should be `Game`, not `Games`.

**Comment by adamj-codethink** [comment]
@barthalion Just doing these changes now, it's strange as i did use https://bilelmoussaoui.github.io/metainfo-generator/#/ to generate this file. So sorry for all these metainfo isses /o\

**Comment by bilelmoussaoui** [comment]
> @barthalion Just doing these changes now, it's strange as i did use https://bilelmoussaoui.github.io/metainfo-generator/#/ to generate this file. So sorry for all these metainfo isses /o\

The tool isn't complete yet o/

**Comment by adamj-codethink** [comment]
@bilelmoussaoui anything i can do to help? I would love to get involved somehow

**Comment by barthalion** [comment]
bot, build com.github.tmewett.BrogueCE

**Comment by adamj-codethink** [comment]
@barthalion I'm not sure why the error is reporting:

 builddir/files/share/appdata/com.github.tmewett.BrogueCE.appdata.xml: FAILED

when my appdata file is now using the metainfo.xml extension?

**Comment by barthalion** [comment]
It still misses releases section. As for filename, flatpak-builder currently renames metainfo files to appdata.

**Comment by adamj-codethink** [comment]
@barthalion ahh ok, none of those generators ask for release, so didn't realise it was needed, i've added the <releases> tag now

**Comment by barthalion** [comment]
bot, build com.github.tmewett.BrogueCE

**Comment by adamj-codethink** [comment]
@barthalion Wooho, everything works/launches, however the games seems to run quite slow and see this error on launch:

`libEGL warning: wayland-egl: could not open /dev/dri/card0 (No such file or directory)`

Any idea what could be, do i need some graphics extension maybe?

**Comment by barthalion** [comment]
Yes, you need `--device=dri`.

**Comment by adamj-codethink** [comment]
Ah odd, i did have that, i must have removed when rebasing, added that now, fingers crossed :D ha

**Comment by barthalion** [comment]
bot, build com.github.tmewett.BrogueCE

**Comment by adamj-codethink** [comment]
@barthalion SUCCESS :smiling_face_with_three_hearts:

**Pull request closed without merging**